src/Xamarin.Android.Build.Tasks/Utilities/AssemblyCompression.cs

```diff
@@ -111,56 +113,72 @@ static CompressionResult Compress (AssemblyData data, string outputDirectory)
 			return CompressionResult.Success;
 		}
 
-		public static string Compress (
-			TaskLoggingHelper log,
-			ITaskItem assembly,
-			IDictionary<AndroidTargetArch, Dictionary<string, CompressedAssemblyInfo>> compressedAssembliesInfo,
-			string compressedOutputDir)
+		public static bool TryCompress (TaskLoggingHelper log, string sourceAssembly, string destinationAssembly, uint descriptorIndex)
 		{
-			if (bool.TryParse (assembly.GetMetadata ("AndroidSkipCompression"), out bool value) && value) {
-				log.LogDebugMessage ($"Skipping compression of {assembly.ItemSpec} due to 'AndroidSkipCompression' == 'true' ");
-				return assembly.ItemSpec;
+			AssemblyData compressedAssembly = new AssemblyData (sourceAssembly, descriptorIndex);
+			CompressionResult result = Compress (compressedAssembly, destinationAssembly);
+
+			if (result != CompressionResult.Success) {
+				switch (result) {
+					case CompressionResult.EncodingFailed:
+						log.LogMessage ($"Failed to compress {sourceAssembly}");
+						break;
+					case CompressionResult.InputTooBig:
```

Comment:
Reviewing the full code file this does not seem to be enabled anymore? The code which raised this issue is commented out.